diff -Naur texinfo-6.1/Makefile.am texinfo-patched-6.1/Makefile.am
--- texinfo-6.1/Makefile.am	2016-06-24 09:01:40.758113476 +1000
+++ texinfo-patched-6.1/Makefile.am	2016-07-05 14:56:41.692942704 +1000
@@ -66,8 +66,7 @@
   SUBDIRS += info
 endif
 endif
-  SUBDIRS += install-info po po_document tp Pod-Simple-Texinfo texindex util \
-             doc man
+  SUBDIRS += install-info po po_document tp Pod-Simple-Texinfo texindex util
 endif
 
 distclean-local:
diff -Naur texinfo-6.1/Makefile.in texinfo-patched-6.1/Makefile.in
--- texinfo-6.1/Makefile.in	2016-07-05 14:51:09.759956987 +1000
+++ texinfo-patched-6.1/Makefile.in	2016-07-05 14:58:05.506939097 +1000
@@ -109,8 +109,7 @@
 @HAVE_TERMLIBS_TRUE@@TOOLS_ONLY_FALSE@am__append_5 = info
 # DOS/Windows don't need TERMLIBS to build Info
 @HAVE_TERMLIBS_FALSE@@HOST_IS_WINDOWS_TRUE@@TOOLS_ONLY_FALSE@am__append_6 = info
-@TOOLS_ONLY_FALSE@am__append_7 = install-info po po_document tp Pod-Simple-Texinfo texindex util \
-@TOOLS_ONLY_FALSE@             doc man
+@TOOLS_ONLY_FALSE@am__append_7 = install-info po po_document tp Pod-Simple-Texinfo texindex util
 
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -280,7 +279,7 @@
 CTAGS = ctags
 CSCOPE = cscope
 DIST_SUBDIRS = gnulib/lib install-info tp util info @native_tools@ po \
-	po_document Pod-Simple-Texinfo texindex doc man
+	po_document Pod-Simple-Texinfo texindex
 am__DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/config.in \
 	$(top_srcdir)/build-aux/ar-lib $(top_srcdir)/build-aux/compile \
 	$(top_srcdir)/build-aux/config.guess \
@@ -1576,7 +1575,7 @@
 	distuninstallcheck dvi dvi-am html html-am info info-am \
 	install install-am install-data install-data-am install-dvi \
 	install-dvi-am install-exec install-exec-am install-html \
-	install-html-am install-info install-info-am install-man \
+	install-html-am install-info install-info-am \
 	install-pdf install-pdf-am install-ps install-ps-am \
 	install-strip installcheck installcheck-am installdirs \
 	installdirs-am maintainer-clean maintainer-clean-generic \
